src/main/java/ml/shifu/shifu/udf/NormalizeUDF.java

```diff
@@ -69,7 +71,28 @@
     private List<DataPurifier> dataPurifiers;
 
     private boolean hasSegExpression = false;
+
+    /**
+     * When this flag is true, we will do normalization in compact mode.
+     * The normalization output will include 4 kind of columns.
+     *     1. Target column.
+     *     2. Meta column.
+     *     3. Final selected column.
+     *     4. Weight column.
+     * We also will keep the order in the input file. For example, we have 4 columns in input file:
+     *     0: final selected
+     *     1: target column
+     *     2: not selected
+     *     3: meta column
+     * We will get below result for compact norm:
+     *     0: final selected
+     *     1: target column
+     *     3: meta column
+     *     4: weight column
+     * The compact mode cannot work with mtl now.
+     */
```

Comment:
Some issues like starting from 0 or 1, better to leverage javadoc table html tags